File: near/src/main/java/com/adroitandroid/near/discovery/server/UdpBroadcastListener.java
package com.adroitandroid.near.discovery.server;

import com.adroitandroid.near.model.Host;

import java.util.Set;

public interface UdpBroadcastListener {
    void onServerSetupFailed(Throwable e);

    void onReceiveFailed();

    void onHostsUpdate(Set<Host> currentHosts);
}

File: near/src/main/java/com/adroitandroid/near/discovery/server/UdpBroadcastListener.kt
package com.adroitandroid.near.discovery.server

import com.adroitandroid.near.model.Host

interface UdpBroadcastListener {
    fun onServerSetupFailed(e: Throwable)
    fun onReceiveFailed()
    fun onHostsUpdate(currentHosts: Set<Host>)
}